### PR TITLE
#147 create car accident service

### PR DESCRIPTION
**Pull request by DenysMatiushko293**
Zaolenyv i znovu vidpravyv do master - Dima pidstrahuvav

### CODE REVIEW

File: CrudMicroservice.Domain/ApiModel/RequestApiModels/CarAccidentRequestApiModel.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+
+namespace CrudMicroservice.Domain.ApiModel.RequestApiModels
+{
+    public class CarAccidentRequestApiModel
+    {
+        public CarAccidentRequestApiModel()
+        {
+            RegistrationDateTime = DateTime.Now;
+        }
```
Comment (pinkevmladchy) [inline CrudMicroservice.Domain/ApiModel/RequestApiModels/CarAccidentRequestApiModel.cs:11]: add empty line

File: CrudMicroservice.Domain/Services/CarAccidentService.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            _mapper = mapper;
+            _carAccidentProtocols = carAccidentProtocols;
+        }
+
+        public async Task<ResponseApiModel<HttpStatusCode>> RegistrationCarAccidentProtocol(CarAccidentRequestApiModel data, string inspectorId)
+        {
+            var carAccidentProtocol = _mapper.Map<CarAccident>(data);
+            carAccidentProtocol.InspectorId = inspectorId;
+            await _carAccidentProtocols.CreateAsync(carAccidentProtocol);
```
Comment (pinkevmladchy) [inline CrudMicroservice.Domain/Services/CarAccidentService.cs:31]: add empty line

**Review by pinkevmladchy — APPROVED** [review]
Functionality works correctly!
But maybe we can USE status of protocol in CAEntity model?

**Review by ICHTLAY — APPROVED** [review]

File: CrudMicroservice.Domain/Services/CarAccidentService.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        public async Task<IEnumerable<CarAccidentResponseApiModel>> FindAllCarAccidentProtocolsByInvolvedId(string inspectorId)
+        {
+            var carAccidentProtocols = await _carAccidentProtocols.GetAllByFilterAsync(x => x.InspectorId == inspectorId);
+            
+            if (carAccidentProtocols == null)
+            {
+                throw new RestException(HttpStatusCode.NotFound, Resources.ResourceManager.GetString("CAprotocolNotFound"));
+            }
+            
+            var mappedCAProtocols = _mapper.Map<IEnumerable<CarAccident>, IEnumerable<CarAccidentResponseApiModel>>(carAccidentProtocols);
```
Comment (SashaKorniichuk) [inline CrudMicroservice.Domain/Services/CarAccidentService.cs]: Map only one type

**Review by SashaKorniichuk — CHANGES_REQUESTED** [review]

File: CrudMicroservice.Domain/Services/CarAccidentService.cs (hunk 3 above)
Comment (DenysMatiushko293) [reply CrudMicroservice.Domain/Services/CarAccidentService.cs]: Done)))

File: CrudMicroservice.Data/Entities/CarAccident.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        [JsonProperty("RegistrationDateTime")]
+        public DateTime RegistrationDateTime { get; set; }
+
+        [BsonElement("AddressOfAccident")]
+        [JsonProperty("AddressOfAccident")]
+        public AddressOfAccident Address { get; set; }
+
+        [BsonElement("SideOfAccident")]
+        [JsonProperty("SideOfAccident")]
+        public Side SideOfAccident { get; set; }
```
Comment (ICHTLAY) [inline CrudMicroservice.Data/Entities/CarAccident.cs:35]: I think that can be many participants of accident, so we should store list of sides

**Review by ICHTLAY — CHANGES_REQUESTED** [review]

**Review by SashaKorniichuk — APPROVED** [review]

**Review by SashaKorniichuk — APPROVED** [review]

File: CrudMicroservice.Data/Entities/CarAccident.cs (hunk 4 above)
Comment (DenysMatiushko293) [reply CrudMicroservice.Data/Entities/CarAccident.cs:35]: no - for each participant of car accident issued separate protocol

**Review by ICHTLAY — APPROVED** [review]

**Review by vladtymo — APPROVED** [review]

**Pull request merged**